homeassistant/components/hue_ble/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""Hue BLE integration."""
+
+import logging
+
+from HueBLE import HueBleLight
+
+from homeassistant.components.bluetooth import (
+    async_ble_device_from_address,
+    async_scanner_count,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the integration from a config entry."""
+
+    assert entry.unique_id is not None
+    address = entry.unique_id.upper()
+
+    ble_device = async_ble_device_from_address(hass, address, connectable=True)
+
+    if ble_device is None:
+        count_scanners = async_scanner_count(hass, connectable=True)
+        _LOGGER.debug("Count of BLE scanners: %i", count_scanners)
+
+        if count_scanners < 1:
+            raise ConfigEntryNotReady(
+                "No Bluetooth scanners are available to search for the light."
+            )
+        raise ConfigEntryNotReady("The light was not found.")
+
+    light = HueBleLight(ble_device)
+
+    if not await light.connect() or not await light.poll_state():
+        raise ConfigEntryNotReady("Device found but unable to connect.")
```

Comment:
So, we do a lot of checks here to ensure the light can be found when setting up the integration.
What happens if the light is found now, but connection is then lost, does the `HueBLE` library automatically reconnect?

Reply:
It automatically attempts to reconnect once but if that does not work it marks itself as unavailable, but any command sent to it afterwards will initiate a reconnection attempt.

Reply:
It looks like I had gotten myself confused with how it worked in an earlier iteration.

The way the library currently works is that if it successfully connects and then becomes disconnected without the disconnect method being called it will attempt to reconnect indefinitely. No command is needed to trigger a re-connection attempt, it does it by itself.